x/wasm/internal/types/types.go

```diff
@@ -130,6 +128,24 @@ func (c *ContractInfo) AdminAddr() sdk.AccAddress {
 	return admin
 }
 
+// NewAbsoluteTxPosition gets a block position from the context
+func NewAbsoluteTxPosition(ctx sdk.Context) *AbsoluteTxPosition {
+	// we muxst safely handle nil gas meters
```

Comment:
```suggestion
	// we must safely handle nil gas meters
```

Reply:
Why do we need to do this? This leads to multiple transactions at `TxIndex` of `0` which destroys al guarantees of unique positioning

Reply:
An empty block gas meter can only exist in tests afaik. I am not sure about the genesis block though.
But the sorting in the new GRPC endpoint take also the contract address into account. It uses the new secondary index `<prefix><codeID><created><contractAddr>` so that we have an order.

The legacy rest endpoint was not updated though to use the second index. The old code only looks at `Created` for sorting. I will compare for address when equal `created`.